src/hooks/useStatsig.tsx

```diff
@@ -7,22 +7,18 @@ import {
   useStatsigClient,
 } from '@statsig/react-bindings';
 
-const statsigClient = new StatsigClient(
-  // need to default to empty string or it breaks if no key is supplied
-  import.meta.env.VITE_STATSIG_CLIENT_KEY ?? '',
-  {},
-  {
-    disableLogging: import.meta.env.VITE_DISABLE_STATSIG,
-    disableStorage: import.meta.env.VITE_DISABLE_STATSIG,
-  }
-);
-// TODO: figure out how to init statsig async so we get configs on the first page load without breaking app state
-// https://linear.app/dydx/project/feature-experimentation-6853beb333d7/overview
-statsigClient.initializeSync();
+import { initStatsig } from '@/lib/statsig';
 
-// Once we figure out how to initialize statsig async, this provider will actually have meaning
-// https://linear.app/dydx/project/feature-experimentation-6853beb333d7/overview
 export const StatsigProvider = ({ children }: { children: React.ReactNode }) => {
+  const [statsigClient, setStatsigClient] = useState<StatsigClient | null>(null);
+  useEffect(() => {
+    const init = async () => {
+      const client = await initStatsig();
+      setStatsigClient(client);
+    };
+    init();
+  }, []);
+  if (!statsigClient) return <div>{children}</div>;
```

Comment:
Will consumers fail gracefully if they assume there's provider data on the context and it's not there?

Reply:
yup, for example i can change the provider to never actually return the StatsingProvider
```
if (!statsigClient) return <div>{children}</div>;
  return <div>{children}</div>;
  // return <StatsigProviderInternal client={statsigClient}> {children} </StatsigProviderInternal>;
```

and yet on the App.tsx i can call
```
const x = useStatsigGateValue(StatSigFlags.ffSkipMigration);
console.log('hey', x);
```

which console logs
<img width="121" alt="Screenshot 2024-07-03 at 12 33 02 PM" src="https://github.com/dydxprotocol/v4-web/assets/37092291/5f82b874-f43c-437a-871c-9575d08618e8">


i think the one thing i'm not sure about is if the statsig client fails to initialize. this should never happen though since the initialization happens locally even if the async call fails (i.e. even when i turn off my wifi i can still load v4-web just fine)